src/compas_rrc/common.py

```diff
@@ -179,6 +194,90 @@ def __setitem__(self, item, value):
+        return Configuration(joint_values, joint_types, joint_names)
+
+    def to_configuration(self, robot, group=None):
+        """Convert the ExternalAxes to a :class:`compas.robots.Configuration`, including the unit conversion
+        from mm and degrees to meters and radians.
+
+        Parameters
+        ----------
+        robot : :class:`compas_fab.robots.Robot`
+            The robot to be configured.
+        group : :obj:`str`
+            The name of the group of joints to be included in the ``Configuration``. Optional.
+            Defaults to the ``robot``'s main group name.
+
+        Returns
+        -------
+        :class:`compas.robots.Configuration`
+        """
+        joint_types = robot.get_configurable_joint_types(group)
+        joint_names = robot.get_configurable_joint_names(group)
+        return self.to_configuration_primitive(joint_types, joint_names)
+
+    @classmethod
+    def from_configuration_primitive(cls, configuration, joint_names=None):
+        """Create an instance of ``ExternalAxes`` from a :class:`compas.robots.Configuration`, including the unit
+        conversion from meters and radians to mm and degrees.
+
+        Parameters
+        ----------
+        configuration : :class:`compas.robots.Configuration`
+            The configuration from which to create the ``ExternalAxes`` instance.
+        joint_names : :obj:`list`
+            An optional list of joint names from the ``configuration`` whose corresponding
+            values will fill the ``ExternalAxes`` values.
+
+        Returns
+        -------
+        :class:`compas_rrc.ExternalAxes`
+        """
+        if joint_names:
+            joint_values = [_convert_unit_to_mm_degrees(configuration[name], configuration.type_dict[name]) for name in joint_names]
+        else:
+            joint_values = [_convert_unit_to_mm_degrees(value, type_) for value, type_ in zip(configuration.joint_values, configuration.joint_types)]
+        return cls(joint_values)
+
+    @classmethod
+    def from_configuration(cls, configuration, robot=None, group=None):
+        """Create an instance of ``ExternalAxes`` from a :class:`compas.robots.Configuration`, including the unit
+        conversion from meters and radians to mm and degrees.
+
+        Parameters
+        ----------
+        configuration : :class:`compas.robots.Configuration`
+            The configuration from which to create the ``ExternalAxes`` instance.
+        robot : :class:`compas_fab.robots.Robot`
+            The robot to be configured.  Optional.
+        group : :obj:`str`
+            The name of the group of joints to be included in the ``ExternalAxes``. Optional.
+            Defaults to the ``robot``'s main group name.
+
+        Returns
+        -------
+        :class:`compas_rrc.ExternalAxes`
+        """
+        joint_names = robot.get_configurable_joint_names(group) if robot else []
```

Comment:
should the `else` case be `[]` or `None`?

Reply:
Either has the same effect.  Do you have a preference?

Reply:
Nope, if both work, then it's good as it is ;)